sdks/python/apache_beam/runners/dataflow/dataflow_runner_test.py

```diff
@@ -404,10 +396,6 @@ def test_gbk_then_flatten_input_visitor(self):
     flat = (none_str_pc, none_int_pc) | beam.Flatten()
     _ = flat | beam.GroupByKey()
 
-    # This may change if type inference changes, but we assert it here
-    # to make sure the check below is not vacuous.
```

Comment:
Hmm... this check seems to be vacuous if the element type is already a tuple, right?

Reply:
I added it back in. The DataflowRunner should change the element type and we should test before and after.